common/djangoapps/third_party_auth/models.py

```diff
@@ -347,6 +348,10 @@ class SAMLProviderConfig(ProviderConfig):
     attr_email = models.CharField(
         max_length=128, blank=True, verbose_name="Email Attribute",
         help_text="URN of SAML attribute containing the user's email address[es]. Leave blank for default.")
+    identity_provider_type = models.CharField(
+        max_length=128, blank=False, verbose_name="Identity Provider Type", default=STANDARD_SAML_PROVIDER_KEY,
+        choices=get_saml_idp_choices(), help_text="If using an identity provider with specific behavioral needs, set here."
```

Comment:
This help text doesn't really describe about what a "provider type" -- it's basically says "if you already know what you are doing, go ahead and use this."  😄